### ISSUE TITLE
Eureka module not found

### ISSUE THREAD

**Issue by lucasastro** [kevin218/Eureka#658]
### What happened?

I have followed the Eureka instructions and am about to run my first Quickstart pipeline.

I have an M1 chip and have thus installed Eureka via the conda environment.yml file.
(the first attempt gave an error.)

I have activated the Eureka environment, but on running hte run_eureka.py file from the Quickstart package it seems the eureka module is not installed or activated, see error traceback output.

Could you help me understand what is going on here? 

### Error traceback output

```
(eureka) lucas [ ~/Analysis/JWST/MyFirstEureka ]$ python run_eureka.py 
Traceback (most recent call last):
  File "~/Analysis/JWST/MyFirstEureka/run_eureka.py", line 2, in <module>
    import eureka.lib.plots
ModuleNotFoundError: No module named 'eureka'
```

### What operating system are you using?

Mac OSX Sonoma 14.4.1

### What version of Python are you running?

Python 3.12.3

### What Python packages do you have installed?

```
# packages in environment at /opt/anaconda3/envs/eureka:
#
# Name                    Version                   Build  Channel
affine                    2.4.0              pyhd8ed1ab_0    conda-forge
alabaster                 0.7.16             pyhd8ed1ab_0    conda-forge
anyio                     4.3.0              pyhd8ed1ab_0    conda-forge
aom                       3.9.0                h537b5a7_0    conda-forge
appnope                   0.1.4              pyhd8ed1ab_0    conda-forge
argon2-cffi               23.1.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0           py39h0f82c59_4    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
arviz                     0.17.1           py39hca03da5_0  
asciitree                 0.3.3                      py_2    conda-forge
asdf                      3.2.0              pyhd8ed1ab_0    conda-forge
asdf-astropy              0.6.1              pyhd8ed1ab_0    conda-forge
asdf-coordinates-schemas  0.3.0              pyhd8ed1ab_0    conda-forge
asdf-standard             1.1.1              pyhd8ed1ab_0    conda-forge
asdf-transform-schemas    0.5.0              pyhd8ed1ab_0    conda-forge
asdf-wcs-schemas          0.4.0              pyhd8ed1ab_0    conda-forge
asteval                   0.9.33             pyhd8ed1ab_0    conda-forge
astraeus                  0.3                      pypi_0    pypi
astropy                   6.0.1            py39h373d45f_0    conda-forge
astropy-healpix           1.0.3            py39h161d348_1    conda-forge
astropy-iers-data         0.2024.6.3.0.31.14    pyhd8ed1ab_0    conda-forge
astroquery                0.4.7                    pypi_0    pypi
astroscrappy              1.1.0            py39h4d8bf0d_2    conda-forge
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
async-lru                 2.0.4              pyhd8ed1ab_0    conda-forge
attrs                     23.2.0             pyh71513ae_0    conda-forge
aws-c-auth                0.7.22               hec39e38_2    conda-forge
aws-c-cal                 0.6.14               h5db4892_1    conda-forge
aws-c-common              0.9.19               h99b78c6_0    conda-forge
aws-c-compression         0.2.18               h5db4892_6    conda-forge
aws-c-event-stream        0.4.2               h5eab607_12    conda-forge
aws-c-http                0.8.1               had10953_17    conda-forge
aws-c-io                  0.14.8               hb5a7b21_5    conda-forge
aws-c-mqtt                0.10.4               h78534b8_4    conda-forge
aws-c-s3                  0.5.9                h1755d02_3    conda-forge
aws-c-sdkutils            0.1.16               h5db4892_2    conda-forge
aws-checksums             0.1.18               h5db4892_6    conda-forge
aws-crt-cpp               0.26.9               h03bff2b_0    conda-forge
aws-sdk-cpp               1.11.329             hb37a6d0_3    conda-forge
azure-core-cpp            1.11.1               he231e37_1    conda-forge
azure-identity-cpp        1.6.0                hd1853d3_1    conda-forge
azure-storage-blobs-cpp   12.10.0              h2ffa867_1    conda-forge
azure-storage-common-cpp  12.5.0               h09a5875_4    conda-forge
babel                     2.14.0             pyhd8ed1ab_0    conda-forge
backports-tarfile         1.2.0                    pypi_0    pypi
batman-package            2.4.9            py39h4d8bf0d_1    conda-forge
bayesicfitting            3.2.1                    pypi_0    pypi
beautifulsoup4            4.12.3             pyha770c72_0    conda-forge
bleach                    6.1.0              pyhd8ed1ab_0    conda-forge
blosc                     1.21.5               h9c252e8_1    conda-forge
bokeh                     2.4.3              pyhd8ed1ab_3    conda-forge
bottleneck                1.3.8            py39h373d45f_0    conda-forge
brotli                    1.1.0                hb547adb_1    conda-forge
brotli-bin                1.1.0                hb547adb_1    conda-forge
brotli-python             1.1.0            py39hb198ff7_1    conda-forge
brunsli                   0.1                  h9f76cd9_0    conda-forge
bzip2                     1.0.8                h93a5062_5    conda-forge
c-ares                    1.28.1               h93a5062_0    conda-forge
c-blosc2                  2.14.4               ha57e6be_1    conda-forge
ca-certificates           2024.6.2             hf0a4a13_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cairo                     1.18.0               hd1e100b_0    conda-forge
ccdproc                   2.4.2              pyhd8ed1ab_0    conda-forge
celerite                  0.4.2            py39h998126f_5    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0           py39he153c15_0    conda-forge
cfitsio                   4.4.0                h808cd33_1    conda-forge
cftime                    1.6.3                    pypi_0    pypi
charls                    2.4.2                h13dd4ca_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
click                     8.1.7           unix_pyh707e725_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               3.0.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.2.2              pyhd8ed1ab_0    conda-forge
contourpy                 1.2.1            py39h48c5dd5_0    conda-forge
corner                    2.2.2              pyhd8ed1ab_0    conda-forge
coverage                  7.5.3            py39hfea33bf_0    conda-forge
crds                      11.17.22                 pypi_0    pypi
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
cytoolz                   0.12.3           py39h17cfd9d_0    conda-forge
dask                      2024.5.2           pyhd8ed1ab_0    conda-forge
dask-core                 2024.5.2           pyhd8ed1ab_0    conda-forge
dask-expr                 1.1.2              pyhd8ed1ab_0    conda-forge
dav1d                     1.2.1                hb547adb_0    conda-forge
debugpy                   1.8.1            py39hf3050f2_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
distributed               2024.5.2           pyhd8ed1ab_0    conda-forge
docutils                  0.20.1                   pypi_0    pypi
drizzle                   1.15.1                   pypi_0    pypi
dynesty                   2.1.2              pyhd8ed1ab_0    conda-forge
emcee                     3.1.6              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
eureka                    0.10                     pypi_0    pypi
exceptiongroup            1.2.0              pyhd8ed1ab_2    conda-forge
executing                 2.0.1              pyhd8ed1ab_0    conda-forge
exotic-ld                 3.2.0                    pypi_0    pypi
expat                     2.6.2                hebf3989_0    conda-forge
fasteners                 0.17.3             pyhd8ed1ab_0    conda-forge
filelock                  3.14.0                   pypi_0    pypi
flake8                    7.0.0              pyhd8ed1ab_0    conda-forge
fmt                       10.2.1               h2ffa867_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_2    conda-forge
fontconfig                2.14.2               h82840c6_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.53.0           py39hfea33bf_0    conda-forge
fqdn                      1.5.1              pyhd8ed1ab_0    conda-forge
freetype                  2.12.1               hadb7bae_2    conda-forge
freexl                    2.0.0                hfbad9fb_0    conda-forge
fsspec                    2024.6.0           pyhff2d567_0    conda-forge
future                    1.0.0              pyhd8ed1ab_0    conda-forge
george                    0.4.0            py39h998126f_7    conda-forge
geos                      3.12.1               h965bd2d_0    conda-forge
geotiff                   1.7.3                h9521f69_0    conda-forge
gflags                    2.2.2             hc88da5d_1004    conda-forge
giflib                    5.2.2                h93a5062_0    conda-forge
glog                      0.7.0                hc6770e3_0    conda-forge
gwcs                      0.21.0             pyhd8ed1ab_0    conda-forge
h11                       0.14.0             pyhd8ed1ab_0    conda-forge
h2                        4.1.0              pyhd8ed1ab_0    conda-forge
h5netcdf                  1.3.0              pyhd8ed1ab_0    conda-forge
h5py                      3.11.0          nompi_py39h534c8c8_101    conda-forge
hdf4                      4.2.15               h2ee6834_7    conda-forge
hdf5                      1.14.3          nompi_hec07895_103    conda-forge
hpack                     4.0.0              pyh9f0ad1d_0    conda-forge
html5lib                  1.1                      pypi_0    pypi
httpcore                  1.0.5              pyhd8ed1ab_0    conda-forge
httpx                     0.27.0             pyhd8ed1ab_0    conda-forge
hyperframe                6.0.1              pyhd8ed1ab_0    conda-forge
icu                       73.2                 hc8870d7_0    conda-forge
idna                      3.7                pyhd8ed1ab_0    conda-forge
image-registration        0.2.10.dev1+g8a3a723          pypi_0    pypi
imagecodecs               2024.6.1         py39h2abf7dc_0    conda-forge
imageio                   2.34.1             pyh4b66e23_0    conda-forge
imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
importlib-metadata        7.1.0              pyha770c72_0    conda-forge
importlib-resources       6.4.0              pyhd8ed1ab_0    conda-forge
importlib_metadata        7.1.0                hd8ed1ab_0    conda-forge
importlib_resources       6.4.0              pyhd8ed1ab_0    conda-forge
iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge
ipykernel                 6.29.3             pyh3cd1d5f_0    conda-forge
ipython                   8.18.1             pyh707e725_3    conda-forge
ipywidgets                8.1.3              pyhd8ed1ab_0    conda-forge
isoduration               20.11.0            pyhd8ed1ab_0    conda-forge
jaraco-classes            3.4.0                    pypi_0    pypi
jaraco-context            5.3.0                    pypi_0    pypi
jaraco-functools          4.0.1                    pypi_0    pypi
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.4              pyhd8ed1ab_0    conda-forge
jmespath                  1.0.1              pyhd8ed1ab_0    conda-forge
joblib                    1.4.2              pyhd8ed1ab_0    conda-forge
json-c                    0.17                 h40ed0f5_0    conda-forge
json5                     0.9.25             pyhd8ed1ab_0    conda-forge
jsonpointer               2.4              py39h2804cbe_3    conda-forge
jsonschema                4.22.0             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2023.12.1          pyhd8ed1ab_0    conda-forge
jsonschema-with-format-nongpl 4.22.0             pyhd8ed1ab_0    conda-forge
jupyter                   1.0.0             pyhd8ed1ab_10    conda-forge
jupyter-lsp               2.2.5              pyhd8ed1ab_0    conda-forge
jupyter_client            8.6.2              pyhd8ed1ab_0    conda-forge
jupyter_console           6.6.3              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2            py39h2804cbe_0    conda-forge
jupyter_events            0.10.0             pyhd8ed1ab_0    conda-forge
jupyter_server            2.14.1             pyhd8ed1ab_0    conda-forge
jupyter_server_terminals  0.5.3              pyhd8ed1ab_0    conda-forge
jupyterlab                4.2.1              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.3.0              pyhd8ed1ab_1    conda-forge
jupyterlab_server         2.27.2             pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        3.0.11             pyhd8ed1ab_0    conda-forge
jwst                      1.11.4                   pypi_0    pypi
jxrlib                    1.1                  h93a5062_3    conda-forge
kealib                    1.5.3                h848a2d4_1    conda-forge
keyring                   25.2.1                   pypi_0    pypi
kiwisolver                1.4.5            py39hbd775c9_1    conda-forge
krb5                      1.21.2               h92f50d5_0    conda-forge
lazy_loader               0.4                pyhd8ed1ab_0    conda-forge
lcms2                     2.16                 ha0e7c42_0    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libabseil                 20240116.2      cxx17_hebf3989_0    conda-forge
libaec                    1.1.3                hebf3989_0    conda-forge
libarchive                3.7.4                h83d404f_0    conda-forge
libarrow                  16.1.0           h28dd788_6_cpu    conda-forge
libarrow-acero            16.1.0           h00cdb27_6_cpu    conda-forge
libarrow-dataset          16.1.0           h00cdb27_6_cpu    conda-forge
libarrow-substrait        16.1.0           hc68f6b8_6_cpu    conda-forge
libavif16                 1.0.4                hb18d617_4    conda-forge
libblas                   3.9.0           22_osxarm64_openblas    conda-forge
libboost-headers          1.85.0               hce30654_1    conda-forge
libbrotlicommon           1.1.0                hb547adb_1    conda-forge
libbrotlidec              1.1.0                hb547adb_1    conda-forge
libbrotlienc              1.1.0                hb547adb_1    conda-forge
libcblas                  3.9.0           22_osxarm64_openblas    conda-forge
libcrc32c                 1.1.2                hbdafb3b_0    conda-forge
libcurl                   8.8.0                h7b6f9a7_0    conda-forge
libcxx                    17.0.6               h5f092b4_0    conda-forge
libdeflate                1.20                 h93a5062_0    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libev                     4.33                 h93a5062_2    conda-forge
libevent                  2.1.12               h2757513_1    conda-forge
libexpat                  2.6.2                hebf3989_0    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libgdal                   3.8.5                hb08d262_7    conda-forge
libgfortran               5.0.0           13_2_0_hd922786_3    conda-forge
libgfortran5              13.2.0               hf226fd6_3    conda-forge
libglib                   2.80.2               h535f939_0    conda-forge
libgoogle-cloud           2.24.0               hfe08963_0    conda-forge
libgoogle-cloud-storage   2.24.0               h3fa5b87_0    conda-forge
libgrpc                   1.62.2               h9c18a4f_0    conda-forge
libhwy                    1.1.0                h2ffa867_0    conda-forge
libiconv                  1.17                 h0d3ecfb_2    conda-forge
libintl                   0.22.5               h8fbad5d_2    conda-forge
libjpeg-turbo             3.0.0                hb547adb_1    conda-forge
libjxl                    0.10.2               h07599a0_0    conda-forge
libkml                    1.3.0             h1eb4d9f_1018    conda-forge
liblapack                 3.9.0           22_osxarm64_openblas    conda-forge
libnetcdf                 4.9.2           nompi_h291a7c2_113    conda-forge
libnghttp2                1.58.0               ha4dd798_1    conda-forge
libopenblas               0.3.27          openmp_h6c19121_0    conda-forge
libparquet                16.1.0           hcf52c46_6_cpu    conda-forge
libpng                    1.6.43               h091b4b1_0    conda-forge
libpq                     16.3                 h7afe498_0    conda-forge
libprotobuf               4.25.3               hbfab5d5_0    conda-forge
libre2-11                 2023.09.01           h7b2c953_2    conda-forge
librttopo                 1.1.0               hc8f776e_15    conda-forge
libsodium                 1.0.18               h27ca646_1    conda-forge
libspatialite             5.1.0                h0c1f73d_6    conda-forge
libsqlite                 3.45.3               h091b4b1_0    conda-forge
libssh2                   1.11.0               h7a5bd25_0    conda-forge
libthrift                 0.19.0               h026a170_1    conda-forge
libtiff                   4.6.0                h07db509_3    conda-forge
libutf8proc               2.8.0                h1a8c8d9_0    conda-forge
libwebp-base              1.4.0                h93a5062_0    conda-forge
libxcb                    1.15                 hf346824_0    conda-forge
libxml2                   2.12.7               ha661575_0    conda-forge
libzip                    1.10.1               ha0bc3c6_3    conda-forge
libzlib                   1.2.13               hfb2fe0b_6    conda-forge
libzopfli                 1.0.3                h9f76cd9_0    conda-forge
llvm-openmp               18.1.6               hde57baf_0    conda-forge
lmfit                     1.2.2              pyhd8ed1ab_1    conda-forge
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lz4                       4.3.3            py39hf99b9d6_0    conda-forge
lz4-c                     1.9.4                hb7217d7_0    conda-forge
lzo                       2.10              h93a5062_1001    conda-forge
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.5            py39h17cfd9d_0    conda-forge
matplotlib                3.8.4            py39hdf13c20_2    conda-forge
matplotlib-base           3.8.4            py39h15359f4_2    conda-forge
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
mccabe                    0.7.0              pyhd8ed1ab_0    conda-forge
mdit-py-plugins           0.4.1                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
minizip                   4.0.6                h0615dfd_0    conda-forge
mistune                   3.0.2              pyhd8ed1ab_0    conda-forge
more-itertools            10.2.0                   pypi_0    pypi
msgpack-python            1.0.8            py39ha1e04a5_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
myst-parser               3.0.1                    pypi_0    pypi
nbclient                  0.10.0             pyhd8ed1ab_0    conda-forge
nbconvert                 7.16.4               hd8ed1ab_0    conda-forge
nbconvert-core            7.16.4             pyhd8ed1ab_0    conda-forge
nbconvert-pandoc          7.16.4               hd8ed1ab_0    conda-forge
nbformat                  5.10.4             pyhd8ed1ab_0    conda-forge
nbsphinx                  0.9.4              pyhd8ed1ab_0    conda-forge
ncurses                   6.5                  hb89a1cb_0    conda-forge
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
netcdf4                   1.6.5                    pypi_0    pypi
networkx                  3.2.1              pyhd8ed1ab_0    conda-forge
notebook                  7.2.0              pyhd8ed1ab_0    conda-forge
notebook-shim             0.2.4              pyhd8ed1ab_0    conda-forge
nspr                      4.35                 hb7217d7_0    conda-forge
nss                       3.100                hc6e9f88_0    conda-forge
numcodecs                 0.12.1           py39hbf7db11_1    conda-forge
numpy                     1.23.0           py39h7df2422_0    conda-forge
numpydoc                  1.7.0              pyhd8ed1ab_0    conda-forge
opencv-python-headless    4.10.0.82                pypi_0    pypi
openjpeg                  2.5.2                h9f1df11_0    conda-forge
openssl                   3.3.0                hfb2fe0b_3    conda-forge
orc                       2.0.1                h47ade37_1    conda-forge
overrides                 7.7.0              pyhd8ed1ab_0    conda-forge
packaging                 24.0               pyhd8ed1ab_0    conda-forge
pandas                    2.2.2            py39h998126f_1    conda-forge
pandoc                    3.2                  hce30654_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parsley                   1.3                      pypi_0    pypi
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
partd                     1.4.2              pyhd8ed1ab_0    conda-forge
pcre2                     10.43                h26f9a81_0    conda-forge
pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge
photutils                 1.11.0           py39h373d45f_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.3.0           py39h3352c98_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pixman                    0.43.4               hebf3989_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
platformdirs              4.2.2              pyhd8ed1ab_0    conda-forge
pluggy                    1.5.0              pyhd8ed1ab_0    conda-forge
poppler                   24.04.0              h42742f0_0    conda-forge
poppler-data              0.4.12               hd8ed1ab_0    conda-forge
poppy                     1.1.1                    pypi_0    pypi
postgresql                16.3                 hdfa2ec6_0    conda-forge
proj                      9.4.0                h52fb9d0_1    conda-forge
prometheus_client         0.20.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.42             pyha770c72_0    conda-forge
prompt_toolkit            3.0.42               hd8ed1ab_0    conda-forge
psutil                    5.9.8            py39h17cfd9d_0    conda-forge
pthread-stubs             0.4               h27ca646_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pyarrow                   16.1.0           py39h799a436_1    conda-forge
pyarrow-core              16.1.0          py39he3fd336_1_cpu    conda-forge
pyarrow-hotfix            0.6                pyhd8ed1ab_0    conda-forge
pycodestyle               2.11.1             pyhd8ed1ab_0    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pyerfa                    2.0.1.4          py39h161d348_1    conda-forge
pyflakes                  3.2.0              pyhd8ed1ab_0    conda-forge
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pyobjc-core               10.2             py39hb167abd_0    conda-forge
pyobjc-framework-cocoa    10.2             py39hb167abd_0    conda-forge
pyparsing                 3.1.2              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
pysynphot                 2.0.0                    pypi_0    pypi
pytest                    8.2.1              pyhd8ed1ab_0    conda-forge
pytest-cov                5.0.0              pyhd8ed1ab_0    conda-forge
pytest-doctestplus        1.2.1              pyhd8ed1ab_0    conda-forge
python                    3.9.7           hc0da0df_3_cpython    conda-forge
python-dateutil           2.9.0              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.19.1             pyhd8ed1ab_0    conda-forge
python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge
python-tzdata             2024.1             pyhd8ed1ab_0    conda-forge
python_abi                3.9                      4_cp39    conda-forge
pytz                      2024.1             pyhd8ed1ab_0    conda-forge
pyvo                      1.5.2                    pypi_0    pypi
pywavelets                1.4.1            py39hf4a74a7_1    conda-forge
pyyaml                    6.0.1            py39h0f82c59_1    conda-forge
pyzmq                     26.0.3           py39he7f0319_0    conda-forge
qtconsole-base            5.5.2              pyha770c72_0    conda-forge
qtpy                      2.4.1              pyhd8ed1ab_0    conda-forge
rasterio                  1.3.10           py39h24a9fa5_1    conda-forge
rav1e                     0.6.6                h69fbcac_2    conda-forge
re2                       2023.09.01           h4cba328_2    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
referencing               0.35.1             pyhd8ed1ab_0    conda-forge
reproject                 0.13.0           py39h373d45f_2    conda-forge
requests                  2.32.3             pyhd8ed1ab_0    conda-forge
rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
rpds-py                   0.18.1           py39h0019b8a_0    conda-forge
scikit-image              0.22.0           py39hf8cecc8_2    conda-forge
scikit-learn              1.5.0            py39h3c33c8b_1    conda-forge
scipy                     1.9.3                    pypi_0    pypi
semantic_version          2.10.0             pyhd8ed1ab_0    conda-forge
send2trash                1.8.3              pyh31c8845_0    conda-forge
setuptools                70.0.0             pyhd8ed1ab_0    conda-forge
setuptools-scm            8.1.0              pyhd8ed1ab_0    conda-forge
shapely                   2.0.4            py39h8b557c8_1    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.2.0                hd04f947_1    conda-forge
sniffio                   1.3.1              pyhd8ed1ab_0    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
snuggs                    1.4.7                      py_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.5                pyhd8ed1ab_1    conda-forge
spdlog                    1.13.0               h5fcca99_0    conda-forge
spherical-geometry        1.3.1                    pypi_0    pypi
sphinx                    7.3.7              pyhd8ed1ab_0    conda-forge
sphinx-automodapi         0.17.0             pyh717bed2_1    conda-forge
sphinx-rtd-theme          2.0.0                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.8              pyhd8ed1ab_0    conda-forge
sphinxcontrib-devhelp     1.0.6              pyhd8ed1ab_0    conda-forge
sphinxcontrib-htmlhelp    2.0.5              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jquery      4.1                      pypi_0    pypi
sphinxcontrib-jsmath      1.0.1              pyhd8ed1ab_0    conda-forge
sphinxcontrib-qthelp      1.0.7              pyhd8ed1ab_0    conda-forge
sphinxcontrib-serializinghtml 1.1.10             pyhd8ed1ab_0    conda-forge
sqlite                    3.45.3               hf2abe2d_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
stcal                     1.4.4                    pypi_0    pypi
stdatamodels              1.7.2                    pypi_0    pypi
stpipe                    0.5.1                    pypi_0    pypi
stsci-image               2.3.8                    pypi_0    pypi
stsci-imagestats          1.8.2                    pypi_0    pypi
stsci-stimage             0.2.9                    pypi_0    pypi
svo-filters               0.4.4                    pypi_0    pypi
svt-av1                   2.1.0                h7bae524_0    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tblib                     3.0.0              pyhd8ed1ab_0    conda-forge
terminado                 0.18.1             pyh31c8845_0    conda-forge
threadpoolctl             3.5.0              pyhc1e730c_0    conda-forge
tifffile                  2024.5.22          pyhd8ed1ab_0    conda-forge
tiledb                    2.23.0               h6a2131e_2    conda-forge
tinycss2                  1.3.0              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5083fa2_1    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
toolz                     0.12.1             pyhd8ed1ab_0    conda-forge
tornado                   6.4              py39h17cfd9d_0    conda-forge
tqdm                      4.66.4             pyhd8ed1ab_0    conda-forge
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
tweakwcs                  0.8.7                    pypi_0    pypi
types-python-dateutil     2.9.0.20240316     pyhd8ed1ab_0    conda-forge
typing-extensions         4.12.1               hd8ed1ab_0    conda-forge
typing_extensions         4.12.1             pyha770c72_0    conda-forge
typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge
tzcode                    2024a                h93a5062_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
uncertainties             3.2.0              pyhd8ed1ab_0    conda-forge
unicodedata2              15.1.0           py39h0f82c59_0    conda-forge
uri-template              1.3.0              pyhd8ed1ab_0    conda-forge
uriparser                 0.9.8                h00cdb27_0    conda-forge
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
webcolors                 1.13               pyhd8ed1ab_0    conda-forge
webencodings              0.5.1              pyhd8ed1ab_2    conda-forge
websocket-client          1.8.0              pyhd8ed1ab_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
widgetsnbextension        4.0.11             pyhd8ed1ab_0    conda-forge
wiimatch                  0.3.2                    pypi_0    pypi
xarray                    2024.5.0           pyhd8ed1ab_0    conda-forge
xarray-einstats           0.7.0              pyhd8ed1ab_0    conda-forge
xerces-c                  3.2.5                hf393695_0    conda-forge
xorg-libxau               1.0.11               hb547adb_0    conda-forge
xorg-libxdmcp             1.1.3                h27ca646_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
yaml                      0.2.5                h3422bc3_2    conda-forge
zarr                      2.18.2             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.5                hcc0f68c_4    conda-forge
zfp                       1.0.1                ha8f4885_0    conda-forge
zict                      3.0.0              pyhd8ed1ab_0    conda-forge
zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               hfb2fe0b_6    conda-forge
zlib-ng                   2.0.7                h1a8c8d9_0    conda-forge
zstd                      1.5.6                hb46c0d2_0    conda-forge
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kevin218** [comment]
My initial guess is that `Eureka!` didn't install correctly and that this may be related to your Python version.  I suggest creating a new conda environment using Python 3.9 and watching the output for errors during installation.

**Comment by taylorbell57** [comment]
Yeah, I agree with Kevin, your setup (including the Python version) indicates that something differed from our installation instructions. First delete your old `eureka` environment, and then precisely follow the installation instructions at https://eurekadocs.readthedocs.io/en/latest/installation.html#installing-with-a-conda-environment-yml-file.

**Comment by lucasastro** [comment]
Dear both,Thanks very much for your replies. I will do what you advise (I thought I had done this already but Python appears to have not picked up the right version). Very noob-y question: how do I remove the existing environment, rather than just deactivating it or replacing it?BestLucasOn 4 Jun 2024, at 22:50, Taylor James Bell ***@***.***> wrote:﻿
Yeah, I agree with Kevin, your setup (including the Python version) indicates that something differed from our installation instructions. First delete your old eureka environment, and then precisely follow the installation instructions at https://eurekadocs.readthedocs.io/en/latest/installation.html#installing-with-a-conda-environment-yml-file.

**Comment by taylorbell57** [comment]
```
conda activate base
conda env remove -n eureka
```

**Comment by lucasastro** [comment]
It still doesn't work. 

This output does not bode well:
lucas $ conda create -n eureka python==3.9.7
...
lucas $ conda activate eureka
lucas $ python --version
Python 3.12.3

Also, if I (because of the M1 chip) try installing with the conda environment.yml file, do I still start by doing step 1 (conda create -n eureka python==3.9.7) i.e. creating 2 conda environments...?

**Comment by taylorbell57** [comment]
No, the conda environment.yml file should make its own new environment, so no need to manually create an environment first

**Comment by taylorbell57** [comment]
After you do `conda env remove -n eureka`, do you get an error message if you try to do `conda activate eureka`? It's good if that gives you an error message, but if not, then somehow the environment isn't being deleted

**Comment by lucasastro** [comment]
yes it does give an error, so the removal is working.
In reaction to your earlier comment, I am currently installing with a conda environment.yml file

**Comment by kevin218** [comment]
Try using Anaconda (or similar gui) to create your conda environment. If
you can solve first problem then the rest might work.

**Comment by taylorbell57** [comment]
Can you please copy-paste the outputs from the following commands:
```
which python
python --version
pip show eureka
```

**Comment by lucasastro** [comment]
(eureka) lucas [ ~ ]$ which python
/opt/anaconda3/envs/eureka/bin/python
(eureka) lucas [ ~ ]$ python --version
Python 3.12.3
(eureka) lucas [ ~ ]$ which pip
/opt/anaconda3/envs/eureka/bin/pip
(eureka) lucas [ ~ ]$ pip show eureka
Name: eureka
Version: 0.10
Summary: A data reduction and analysis pipeline intended for time-series observations with JWST.
Home-page: https://github.com/kevin218/Eureka
Author: Eureka! pipeline developers
Author-email: kbstevenson@gmail.com
License: MIT License
Location: /opt/anaconda3/envs/eureka/lib/python3.9/site-packages
Requires: astraeus, astropy, astroquery, batman-package, bokeh, ccdproc, celerite, corner, crds, dynesty, emcee, exotic-ld, george, h5py, lmfit, matplotlib, numpy, pandas, photutils, requests, scipy, setuptools-scm, stdatamodels, svo-filters, tqdm
Required-by:

**Comment by taylorbell57** [comment]
Huh, that is very strange - it seems that Kevin is right, and there is something weird with your Anaconda or Miniconda installation that is somehow forcing the use of `Python==3.12.3`... I can't really help troubleshoot that, but I recommend fiddling around with your conda installation until you can create an environment with your desired Python version. And perhaps look for people having similar issues on Stack Exchange or websites like that?

**Comment by lucasastro** [comment]
OK, I will try this and also seek advice.
Thanks for the Anaconda suggestion, and for your help so far. 
Will update this issue when it is resolved, may be useful to the community.

**Comment by lucasastro** [comment]
It still gives the same error.

I deactivated and removed existing Eureka environment, including the Eureka folder on the disk
I reinstalled Python 3.9.7 from python.org, removed mentions of 3.12 from $PATH. 
Then I reinstalled Eureka following the instructions (through .yml file).
Now it does work in the correct Python version but the error persists.
Any ideas what may be wrong? I have a feeling that the solution may be simpler than thought before...

I can of course remove and reinstall Anaconda, but it seems to me that the Python version is not the factor causing the problem. 

Tomorrow I will retry by more robust reinstalls, but perhaps some nugget of wisdom already comes to mind...

Output:

(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ python run_eureka.py
Traceback (most recent call last):
  File "/Users/lucas/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka/run_eureka.py", line 2, in <module>
    import eureka.lib.plots
ModuleNotFoundError: No module named 'eureka'
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ which python
/opt/anaconda3/envs/eureka/bin/python
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ python --version
Python 3.9.7
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ which pip
/opt/anaconda3/envs/eureka/bin/pip
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ pip show eureka
Name: eureka
Version: 0.10
Summary: A data reduction and analysis pipeline intended for time-series observations with JWST.
Home-page: https://github.com/kevin218/Eureka
Author: Eureka! pipeline developers
Author-email: kbstevenson@gmail.com
License: MIT License
Location: /opt/anaconda3/envs/eureka/lib/python3.9/site-packages
Requires: astraeus, astropy, astroquery, batman-package, bokeh, ccdproc, celerite, corner, crds, dynesty, emcee, exotic-ld, george, h5py, lmfit, matplotlib, numpy, pandas, photutils, requests, scipy, setuptools-scm, stdatamodels, svo-filters, tqdm
Required-by:

**Comment by taylorbell57** [comment]
Hmm, I'm really at a loss here... Some things come to mind to help me troubleshoot this more.

1. Can you show me the terminal inputs and outputs for when you're installing the package? I want to see if there is some kind of subtle error message or something that'd help me better understand this.
2. Can you tell me what is on line 1 of your `run_eureka.py` file?
3. If you navigate to the Eureka folder that was produced from the `git clone` command, what do you get when you run `pytest tests`
5. Beyond that, you could also try upgrading to the `main` branch of Eureka! which has several updates (perhaps one of which will resolve your problem) but has a bit higher a risk for bugs in the code. To use the main branch instead, replace the instance of `v0.10` with `main` when running the `git clone` command

**Comment by taylorbell57** [comment]
I am so deeply confused... Your `pip show eureka` command works and shows that eureka is installed, and your `pytest tests` command works (although that actually doesn't use your installed eureka and instead uses the raw code in the downloaded GitHub folder, so at least we know the issue isn't with any of your other dependencies). I initially suspected the issue could be that you're running the update steps right after installing (which you should definitely not do btw), but you didn't do that in the `Install_Eureka_main_output.txt` file and still got the error so I no longer suspect that's the cause. I see your `conda` has updates available which could possibly be the issue (could be worth trying their recommended `conda update -n base -c defaults conda` command first) but I kinda doubt that'll solve anything. The last thing I can think of is that something is wrong with your `PATH` environment variable - do you mind copy-pasting the outputs of running `echo $PATH` while your eureka conda environment is active?

**Comment by lucasastro** [comment]
Indeed, updating conda did not change the outcome.

The output for $PATH while in Eureka:
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ echo $PATH                                       
/Library/Frameworks/Python.framework/Versions/3.9/bin:/opt/anaconda3/envs/eureka/bin:/Library/Frameworks/Python.framework/Versions/3.9/bin:/Users/lucas/DS9:/Users/lucas/anaconda/bin:/opt/local/bin:/opt/local/sbin:/Library/Frameworks/Python.framework/Versions/Current/bin:/Applications/scisoft/i386/bin:/Applications/scisoft/all/bin:/bin:/usr/bin:/Users/lucas/p2pp-3.3.0/bin:/Users/lucas/10FEBpl1.1/system/unix:/Users/lucas/gcc-4.5.0:/Applications/scisoft/i386/bin:/Applications/scisoft/all/bin:/Applications/rsi/idl:/Applications/rsi/idl/bin:/Users/Lucas/idl/idl_startup.pro:/Users/Lucas/idl/programs:/Applications/itt/idl70/lib:/Applications/itt/idl70/examples:/Users/lucas/gasgano/bin:/Users/lucas/p2pp-3.3.0/bin:/usr/local/bin:/usr/texbin:/Library/TeX/texbin:/Users/lucas/latexdiff:/Users/lucas/cires/bin:/Applications/ImageJ   

There's a lot of stuff in there that I never use anymore (idl, gasgano, p2pp...) but it looks harmless. Although I remember that scisoft contained a Python installation.

Perhaps a stupid remark, but in /opt/anaconda3/envs/eureka/bin there is no file called 'eureka'

Note that (see https://github.com/kevin218/Eureka/issues/658#issuecomment-2153139020) I cleaned up $PATH a bit when manually installing Python 3.9 (since it somehow kept refusing to adapt it from the environment). This worked:
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ python --version
Python 3.9.7

Note also that I never used a GUI to do the installation, but have run everything from a Terminal. So there is still that option. I have previously installed Anaconda and a (expired) trial version of PyCharm.

Is it a Eureka! moment yet...?

**Comment by taylorbell57** [comment]
Hmm, I suspect that `/Library/Frameworks/Python.framework` folder is the root of this issue. I think you likely installed python using `brew` or some other installer at some point, and because that is at the start of the `PATH` variable, I think it might be causing issues (see [this stackoverflow post](https://stackoverflow.com/questions/22255579/homebrew-brew-doctor-warning-about-library-frameworks-python-framework-even-wi) where some other folks were having weird issues caused by that python being installed). Can you try either removing that folder from your `PATH` variable and/or moving it after the `/opt/anaconda3/envs/eureka/bin` folder?

**Comment by lucasastro** [comment]
:,-(

I removed this reference from $PATH:
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ echo $PATH
/opt/miniconda3/envs/eureka/bin:/Users/lucas/DS9:/Users/lucas/anaconda/bin:/opt/local/bin:/opt/local/sbin:/Applications/scisoft/i386/bin:/Applications/scisoft/all/bin:/bin:/usr/bin:/Users/lucas/p2pp-3.3.0/bin:/Users/lucas/10FEBpl1.1/system/unix:/Users/lucas/gcc-4.5.0:/Applications/scisoft/i386/bin:/Applications/scisoft/all/bin:/Applications/rsi/idl:/Applications/rsi/idl/bin:/Users/Lucas/idl/idl_startup.pro:/Users/Lucas/idl/programs:/Applications/itt/idl70/lib:/Applications/itt/idl70/examples:/Users/lucas/gasgano/bin:/Users/lucas/p2pp-3.3.0/bin:/usr/local/bin:/usr/texbin:/Library/TeX/texbin:/Users/lucas/latexdiff:/Users/lucas/cires/bin:/Applications/ImageJ

(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ python run_eureka.py
Traceback (most recent call last):
  File "/Users/lucas/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka/run_eureka.py", line 2, in <module>
    import eureka.lib.plots
ModuleNotFoundError: No module named 'eureka'

Curiously, by the way, I got this error:
(base) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/Eureka ]$ conda env create --file environment.yml --force
usage: conda [-h] [-v] [--no-plugins] [-V] COMMAND ...
conda: error: unrecognized arguments: --force

This happened before, and like now I typed the command without --force. 
Earlier, it did accept --force but only after I changed the Python version to 3.9.7.
But I guess this is no biggie.

I can get rid of more redundant $PATH variables but it feels a bit desparate.
Any ideas?

**Comment by lucasastro** [comment]
A possible lead:
Some creative Googling let me to try the following:

(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ python
Python 3.9.7 (v3.9.7:1016ef3790, Aug 30 2021, 16:25:35) 
[Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> print(sys.path)
['', '/Library/Frameworks/Python.framework/Versions/3.9/lib/python39.zip', '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9', '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload', '/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages']

I don't think this path includes the directory where eureka is installed... should it?

**Comment by taylorbell57** [comment]
Huh that's very strange - yes it seems the python being used is still not the conda python (since that isn't listed anywhere in the `sys.path`). I have no idea how that could be the case if `which python` says it'll use the conda python, but it seems clear that it isn't. My `sys.path` list (on Windows Subsystem for Linux, so the paths are formatted differently but give you an idea of what the `sys.path` should look like) is:
```
['/mnt/wsl/ubuntuSSD/home/taylor/miniconda3/envs/eureka_starry/bin', 
'/home/taylor/miniconda3/envs/eureka_starry/lib/python39.zip',
'/home/taylor/miniconda3/envs/eureka_starry/lib/python3.9', 
'/home/taylor/miniconda3/envs/eureka_starry/lib/python3.9/lib-dynload', '', 
'/home/taylor/miniconda3/envs/eureka_starry/lib/python3.9/site-packages', '/home/taylor/Eureka/src']
```

That second-to-last `.../python3.9/site-packages` folder is where eureka is installed

**Comment by lucasastro** [comment]
Bulls eye. 
Somehow 'python' calls a different (fossilized..?) Python than the one specified by 'which python'. The welcome message looks markedly different. And if I explicitly call the 'conda python' calling Eureka modules works:

(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ which python
/opt/miniconda3/envs/eureka/bin/python
(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ python
Python 3.9.7 (v3.9.7:1016ef3790, Aug 30 2021, 16:25:35) 
[Clang 12.0.5 (clang-1205.0.22.11)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import eureka.lib.plots
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'eureka'
>>> ^D

(eureka) lucas [ ~/Documents/ScienceHD/REVEAL_HD/Analysis/JWST/MyFirstEureka ]$ /opt/miniconda3/envs/eureka/bin/python
Python 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:22:19) 
[Clang 11.1.0 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> help(eureka)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'eureka' is not defined
>>> import eureka.lib.plots
Could not import starry and/or pymc3 related packages. Functionality may be limited.
>>> 

So, this is a 'fix' to the problem, although I would like the alias 'python' to call the conda version when in eureka. 
What do you think may be places to look for this bodged alias?

**Comment by taylorbell57** [comment]
Whew, it was a challenge but I'm glad we got this sorted out. I'll close this issue now since you've sorted it out and it wasn't something caused on our end, but please do open a new issue if you encounter a bug with Eureka! now that you're actually able to use the code. And this closed issue will still remain archived here on GitHub to help any future people that encounter this issue

**Comment by lucasastro** [comment]
Understood. Thanks for your time and effort!!On 12 Jun 2024, at 00:49, Taylor James Bell ***@***.***> wrote:﻿
Whew, it was a challenge but I'm glad we got this sorted out. I'll close this issue now since you've sorted it out and it wasn't something caused on our end, but please do open a new issue if you encounter a bug with Eureka! now that you're actually able to use the code. And this closed issue will still remain archived here on GitHub to help any future people that encounter this issue
